packages/server/src/utilities/security.ts

```diff
@@ -8,14 +9,16 @@ export const CURRENTLY_SUPPORTED_LEVELS: string[] = [
 ]
 
 export function getPermissionType(resourceId: string) {
-  const docType = Object.values(DocumentType).filter(docType =>
-    resourceId.startsWith(docType)
-  )[0]
```

Comment:
There's some light refactoring in this file because my implementation originally went in here but I later switched tactics. I did, however, discover a bug in this code here. Because `DocumentType`s can overlap (e.g. `datasource_` is a subset of `datasource_plus_`), you need to search through this list in order of longest to shortest. So I extracted a helper to do just that.